src/sentry/sentry_metrics/querying/metadata/metrics_correlations.py

```diff
@@ -413,22 +458,17 @@ def supports(cls, metric_mri: str) -> bool:
     def _get_segments(
         self,
         metric_mri: str,
-        conditions: Optional[ConditionGroup],
+        conditions: QueryConditions,
         start: datetime,
         end: datetime,
         min_value: Optional[float],
         max_value: Optional[float],
     ) -> Sequence[Segment]:
-        where = []
+        where: List[QueryCondition] = []
 
-        transformed_conditions = transform_conditions_to_tags(
-            conditions=conditions, check_sentry_tags=True
-        )
-        transformed_conditions = transform_conditions_with(
-            conditions=transformed_conditions, mappings=SENTRY_TAG_TO_COLUMN_NAME
-        )
-        if transformed_conditions:
-            where += transformed_conditions
+        conditions.add_visitor(TagsTransformationVisitor(check_sentry_tags=True))
+        conditions.add_visitor(MappingTransformationVisitor(mappings=SENTRY_TAG_TO_COLUMN_NAME))
+        where += conditions.get()
```

Comment:
The new visitors implementation is lazy, so at the end the actual conditions are computed.